ioctl/cmd/account/accountcreate2.go

```diff
@@ -0,0 +1,239 @@
+	k := ecrypto.FromECDSAPub(publicKey)
+	h := hash.Hash160b(k[1:])
+	return h[:]
+}
+
+func init() {
+	accountCreateByMnemonicCmd.Flags().UintVarP(&numAccounts, "num", "n", 1,
+		config.TranslateInLang(flagNumUsages, config.UILanguage))
+}
+
+func createUserByMnemonic() error {
+	output.PrintQuery("Set password\n")
+	password, err := util.ReadSecretFromStdin()
+	if err != nil {
+		return output.NewError(output.InputError, "failed to get password", err)
+	}
+	output.PrintQuery("Enter password again\n")
+	passwordAgain, err := util.ReadSecretFromStdin()
+	if err != nil {
+		return output.NewError(output.InputError, "failed to get password", err)
+	}
+	if password != passwordAgain {
+		return output.NewError(output.ValidationError, ErrPasswdNotMatch.Error(), nil)
+	}
+
+	entropy, _ := bip39.NewEntropy(128)
+	mnemonic, _ := bip39.NewMnemonic(entropy)
+
+	output.PrintResult(fmt.Sprintf("Mnemonic pharse: %s\n"+
+		"Save them somewhere safe and secret.", mnemonic))
+
+	privateKey, err := rsa.GenerateKey(rand.Reader, 2048)
+	if err != nil {
+		return err
+	}
+
+	publicKey := privateKey.PublicKey
+
+	encryptedMnemonic, err := rsa.EncryptOAEP(
+		sha256.New(),
+		rand.Reader,
+		&publicKey,
+		[]byte(mnemonic),
+		nil)
+	if err != nil {
+		return err
+	}
+
+	encryptedPassword, err := rsa.EncryptOAEP(
+		sha256.New(),
+		rand.Reader,
+		&publicKey,
+		[]byte(password),
+		nil)
+	if err != nil {
+		return err
+	}
+
+	out, err := yaml.Marshal(&mnemonicConfig{
+		PrivateKey: x509.MarshalPKCS1PrivateKey(privateKey),
+		Mnemonic:   encryptedMnemonic,
+		Password:   encryptedPassword,
+	})
+	if err != nil {
+		return output.NewError(output.SerializationError, "failed to marshal config", err)
+	}
+	if err := ioutil.WriteFile(mnemonicConfigFilePath, out, 0600); err != nil {
+		return output.NewError(output.WriteFileError,
+			fmt.Sprintf("failed to write to config file %s", mnemonicConfigFilePath), err)
+	}
+
+	return nil
+}
+
+func readMnemonicConfig() (*mnemonicConfig, error) {
+	var cfg *mnemonicConfig
+	out, err := ioutil.ReadFile(mnemonicConfigFilePath)
+	if err != nil {
+		return nil, err
+	}
+	if err = yaml.Unmarshal(out, &cfg); err != nil {
+		return nil, err
+	}
+	return cfg, nil
+}
+
+func accountCreateByMnemonic() error {
+	var err error
+	newAccounts := make([]generatedAccount, 0)
+
+	if !fileutil.FileExists(mnemonicConfigFilePath) {
+		return createUserByMnemonic()
+	}
+
+	rmc, err := readMnemonicConfig()
+	mnemonic, err := rmc.DecryptMnemonic()
+	if err != nil {
+		return err
+	}
+
+	password, err := rmc.DecryptPassword()
+	if err != nil {
+		return err
+	}
+	output.PrintResult(fmt.Sprintf("mnemonic:%s\npassword:%s", mnemonic, password))
+
+	wallet, err := hdwallet.NewFromMnemonic(string(mnemonic))
+	if err != nil {
+		return err
+	}
+
+	var i uint32
+	for i = rmc.ID; i < rmc.ID+uint32(numAccounts); i++ {
+		//https://gowalker.org/github.com/ethereum/go-ethereum/accounts#DefaultBaseDerivationPath
+		derivationPath := hdwallet.DefaultRootDerivationPath
```

Comment:
we have registered cointype = 304, see https://github.com/satoshilabs/slips/blob/master/slip-0044.md

@ququzone  can you check/confirm, in our ioPay mnemonic/hdwallet implementation, do we use 304 correctly?